cairo-contracts/packages/apps/src/transfer/components/transfer.cairo

```diff
@@ -289,7 +289,10 @@ pub mod TokenTransferComponent {
             ref self: ComponentState<TContractState>, denom: PrefixedDenom,
         ) -> ContractAddress {
             let mut token = self.get_token(denom.key());
-            assert(token.is_zero(), TransferErrors::TOKEN_ALREADY_EXISTS);
+
+            if !token.is_zero() {
+                return token.address;
```

Comment:
Here I make token creation to be no-op if a token already exists. This helps resolving the race condition where we try to create a token while a relayer is relaying and triggering the token creation at the same time.

Reply:
The race condition makes sense. It's alright for me.